<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <!-- jQuery -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
  
  <!-- bootstrap -->
  <!--<link href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet"  id="style">-->
  <script src="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
  
  <!-- highlight.js -->
  <!--<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" rel="stylesheet" id="code-style">-->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/highlight.min.js"></script>
  <script>
  hljs.LANGUAGES.r=function(a){var b="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[a.HCM,{b:b,l:b,k:{keyword:"function if in break next repeat else for return switch while try tryCatch|10 stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...|10",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[a.BE],r:0},{cN:"string",b:"'",e:"'",c:[a.BE],r:0}]}}(hljs); </script>
  <!--<script type="text/javascript", src="https://yandex.st/highlightjs/7.3/languages/r.min.js"></script>-->
  
  <!-- Manific Popup -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/magnific-popup.js/0.8.9/jquery.magnific-popup.min.js"></script>
  
  <script type="text/javascript"
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>
  <script defer="defer">
  // Function to generate the dynamic table of contents
  jQuery.fn.generate_TOC = function () {
    var base = $(this[0]);
  
    var selectors = ['h1', 'h2', 'h3', 'h4'];
  
    var last_ptr = [{}, {}, {}, {}];
  
    var anchors = {};
  
    generate_anchor = function (text) {
      var test = text.replace(/\W/g, '_');
  
      while(test in anchors){
        //if no suffix, add one
        if(test.match(/_\d+$/) === null){
          test = test + "_2";
        }
        //else generate unique id for duplicates by adding one to the suffix
        else {
          test = test.replace(/_(\d+)$/, function(match, number){ var num=+number+1; return("_" + num) });
        }
      }
      anchors[test]=1;
      return(test);
    }
  
    $(selectors.join(',')).filter(function(index) { return $(this).parent().attr("id") != 'header'; }).each(function () {
  
      var heading = $(this);
      var idx = selectors.indexOf(heading.prop('tagName').toLowerCase());
      var itr = 0;
  
      while (itr <= idx) {
        if (jQuery.isEmptyObject(last_ptr[itr])) {
          last_ptr[itr] = $('<ul>').addClass('nav');
          if (itr === 0) {
            base.append(last_ptr[itr])
          } else {
            if(last_ptr[itr-1].children('li').length === 0){
              last_ptr[itr-1].append(last_ptr[itr]);
            }
            else {
              last_ptr[itr - 1].children('li').last().append(last_ptr[itr]);
            }
          }
        }
        itr++;
      }
      var anchor = generate_anchor(heading.text());
      heading.attr('id', anchor);
      var a = $('<a>')
      .text(heading.text())
      .attr('href', '#' + anchor);
  
    var li = $('<li>')
      .append(a);
  
    last_ptr[idx].append(li);
    for (i = idx + 1; i < last_ptr.length; i++) {
      last_ptr[i] = {};
    }
    });
  }
  /* run scripts when document is ready */
  $(function() {
    "use strict";
  
    var $window = $(window);
    var $body = $(document.body);
  
    /* size of thumbnails */
  
    var hidden_types = ['source']
    var output_types = ['output', 'message', 'warning', 'error']
  
    /* style tables */
    $('table').addClass('table table-striped table-bordered table-hover table-condensed');
  
    $('pre code').each(function(i, e) {
      hljs.highlightBlock(e);
    });
  
    /* Magnific Popup */
    $(".thumbnail").each(function(){
      $(this).magnificPopup({
        disableOn: 768,
        closeOnContentClick: true,
  
        type: 'image',
        items: {
          src: $(this).find('img').attr('src'),
        }
      });
    });
  
    function toggle_block(obj, show) {
      var span = obj.find('span');
      if(show === true){
        span.removeClass('glyphicon-chevron-up').addClass('glyphicon-chevron-down');
        obj.next('pre').slideDown();
      }
      else {
        span.removeClass('glyphicon-chevron-down').addClass('glyphicon-chevron-up');
        obj.next('pre').slideUp();
      }
    }
  
    function toggle_thumbnails(imgs, show){
      if(show === true){
        imgs.parents().show()
        imgs.slideDown();
      }
      else {
        imgs.slideUp(400, function(){ $(this).parent().hide(); });
      }
    }
  
    function global_toggle(obj){
      var type = obj.attr('type');
      var show = !obj.parent('li').hasClass('active');
      if(show === true){
        obj.parent('li').addClass('active');
      }
      else{
        obj.parent('li').removeClass('active');
      }
      if(type == 'figure'){
        toggle_thumbnails($('.thumbnail img'), show);
      }
      else {
        $('.toggle.' + type).each(function() { toggle_block($(this), show); });
      }
    }
  
    /* onclick toggle next code block */
    $('.toggle').click(function() {
      var span = $(this).find('span');
      toggle_block($(this), !span.hasClass('glyphicon-chevron-down'));
      return false
    })
  
    // global toggles
    $('.toggle-global').click(function(){
      var type = $(this).attr('type');
      if(type === 'all-source'){
          $('li a.source').each(function() {
            global_toggle($(this));
          });
        }
      else if(type === 'all-output'){
        $.each(output_types, function(i, val){
          console.log(val);
          global_toggle($('li a.' + val));
        });
      }
      else {
        console.log($(this));
        global_toggle($(this));
      }
      return false;
    });
    /* table of contents */
    if($(['h1', 'h2', 'h3', 'h4'].join(',')).length > 0){
      $('body > #wrap > .container > .row').append('<div class="col-md-2"><div id="toc" class="well sidebar sidenav affix hidden-print"/></div>');
      $('#toc').generate_TOC();
    }
  
    $.each(hidden_types, function(i, type) {
      $('li[type=' + type + ']').each(function(){ global_toggle($(this)); });
    });
  
    /* remove paragraphs with no content */
    $('p:empty').remove();
  
    $body.scrollspy({
      target: '.sidebar',
    });
  
    /* theme switch */
    $('.theme-switch').click(function(){
      var css = $('link[title=' + $(this).attr('title') + ']');
      $('#theme[rel=stylesheet]').attr('href', css.attr('href'));
      $('.theme-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
    /* code style switch */ //TODO use same function for both of these?
    $('.highlight-switch').click(function(){
      var css = $('link[title="' + $(this).attr('title') + '"]');
      $('#highlight[rel=stylesheet]').attr('href', css.attr('href'));
      $('.highlight-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
  
    //TODO refresh on show/hide
    $window.on('load', function () {
      $body.scrollspy('refresh');
    })
  
  });
  
  </script>
  <style>
  /* Knitr_bootstrap styles */
  #header {
    display: none !important;
    visibility: hidden !important;
  }
  #wrap .container-fluid {
    padding: 0;
    overflow: hidden;
  }
  .toggle{
    text-transform: capitalize;
  }
  
  .toggle-global{
    text-transform: capitalize;
  }
  
  /* Sticky footer styles */
  * {
    margin:0;
  }
  html,
  body {
      height: 100%;
      padding:0 !important;
      /* The html and body elements cannot have any padding or margin. */
      /*overflow-x: hidden;*/
  }
  
  /* Wrapper for page content to push down footer */
  #wrap {
      min-height: 100%;
      height: auto !important;
      height: 100%;
      /* Negative indent footer by it's height */
      margin: 0 auto -120px;
  }
  
  /* Set the fixed height of the footer here */
  #push,
  #footer {
      height: 120px;
  }
  
  #footer {
    text-align: center;
  }
  
  /* Top level subheader elements.  These are the first nested items underneath a header element. */
  .header li {
    font-size: 20px;
  }
  
  /* Makes the font smaller for all subheader elements. */
  .sub-header li {
      font-size: 12px;
  }
  
  button.thumbnails {
    margin-left:0px;
  }
  
  /*
   * Side navigation
   *
   * Scrollspy and affixed enhanced navigation to highlight sections and secondary
   * sections of docs content.
   */
  
  /* By default it's not affixed in mobile views, so undo that */
  .sidebar.affix {
    position: static;
  }
  
  /* First level of nav */
  .sidenav {
    margin-top: 30px;
    margin-bottom: 30px;
    padding-top:    10px;
    padding-bottom: 10px;
    border-radius: 5px;
  }
  
  /* All levels of nav */
  .sidebar .nav > li > a {
    display: block;
    padding: 5px 20px;
  }
  .sidebar .nav > li > a:hover,
  .sidebar .nav > li > a:focus {
    text-decoration: none;
    border-right: 1px solid;
  }
  .sidebar .nav > .active > a,
  .sidebar .nav > .active:hover > a,
  .sidebar .nav > .active:focus > a {
    font-weight: bold;
    background-color: transparent;
    border-right: 1px solid;
  }
  
  /* Nav: second level (shown on .active) */
  .sidebar .nav .nav {
    display: none; /* Hide by default, but at >768px, show it */
    margin-bottom: 8px;
  }
  .sidebar .nav .nav > li > a {
    padding-top:    3px;
    padding-bottom: 3px;
    padding-left: 30px;
    font-size: 90%;
  }
  
  .sidebar .nav .nav .nav > li > a {
    padding-left: 40px;
  }
  .sidebar .nav .nav .nav .nav > li > a {
    padding-left: 50px;
  }
  
  /* Show and affix the side nav when space allows it */
  @media screen and (min-width: 992px) {
    .sidebar .nav > .active > ul {
      display: block;
    }
    /* Widen the fixed sidebar */
    .sidebar.affix,
    .sidebar.affix-bottom {
      width: 213px;
    }
    .sidebar.affix-top,
    .sidebar.affix {
      position: fixed; /* Undo the static from mobile first approach */
      top: 30px;
    }
    .sidebar.affix-bottom {
      position: absolute; /* Undo the static from mobile first approach */
    }
    .sidebar.affix-bottom .sidenav,
    .sidebar.affix .sidenav {
      margin-top: 0;
      margin-bottom: 0;
    }
  }
  @media screen and (min-width: 1200px) {
    /* Widen the fixed sidebar again */
    .sidebar.affix-bottom,
    .sidebar.affix {
      width: 263px;
    }
  }
  
  #toc {
    padding: 10px 0px;
    margin:0;
    border:0;
  }
  
  
  .panel pre {
    margin: 0;
    padding: 0;
    border: 0;
  }
  button + pre {
    margin: 0;
    padding: 0;
  }
  pre code {
    border-radius: 0;
  }
  /* Magnific Popup CSS */
  .mfp-bg {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1042;
    overflow: hidden;
    position: fixed;
    background: #0b0b0b;
    opacity: 0.8;
    filter: alpha(opacity=80); }
  
  .mfp-wrap {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1043;
    position: fixed;
    outline: none !important;
    -webkit-backface-visibility: hidden; }
  
  .mfp-container {
    text-align: center;
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    padding: 0 8px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box; }
  
  .mfp-container:before {
    content: '';
    display: inline-block;
    height: 100%;
    vertical-align: middle; }
  
  .mfp-align-top .mfp-container:before {
    display: none; }
  
  .mfp-content {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    margin: 0 auto;
    text-align: left;
    z-index: 1045; }
  
  .mfp-inline-holder .mfp-content,
  .mfp-ajax-holder .mfp-content {
    width: 100%;
    cursor: auto; }
  
  .mfp-ajax-cur {
    cursor: progress; }
  
  .mfp-zoom-out-cur,
  .mfp-zoom-out-cur .mfp-image-holder .mfp-close {
    cursor: -moz-zoom-out;
    cursor: -webkit-zoom-out;
    cursor: zoom-out; }
  
  .mfp-zoom {
    cursor: pointer;
    cursor: -webkit-zoom-in;
    cursor: -moz-zoom-in;
    cursor: zoom-in; }
  
  .mfp-auto-cursor .mfp-content {
    cursor: auto; }
  
  .mfp-close,
  .mfp-arrow,
  .mfp-preloader,
  .mfp-counter {
    -webkit-user-select: none;
    -moz-user-select: none;
    user-select: none; }
  
  .mfp-loading.mfp-figure {
    display: none; }
  
  .mfp-hide {
    display: none !important; }
  
  .mfp-preloader {
    color: #cccccc;
    position: absolute;
    top: 50%;
    width: auto;
    text-align: center;
    margin-top: -0.8em;
    left: 8px;
    right: 8px;
    z-index: 1044; }
  
  .mfp-preloader a {
    color: #cccccc; }
  
  .mfp-preloader a:hover {
    color: white; }
  
  .mfp-s-ready .mfp-preloader {
    display: none; }
  
  .mfp-s-error .mfp-content {
    display: none; }
  
  button.mfp-close,
  button.mfp-arrow {
    overflow: visible;
    cursor: pointer;
    background: transparent;
    border: 0;
    -webkit-appearance: none;
    display: block;
    padding: 0;
    z-index: 1046;
    -webkit-box-shadow: none;
    box-shadow: none; }
  
  button::-moz-focus-inner {
    padding: 0;
    border: 0; }
  
  .mfp-close {
    width: 44px;
    height: 44px;
    line-height: 44px;
    position: absolute;
    right: 0;
    top: 0;
    text-decoration: none;
    text-align: center;
    opacity: 0.65;
    padding: 0 0 18px 10px;
    color: white;
    font-style: normal;
    font-size: 28px;
    font-family: Arial, Baskerville, monospace; }
    .mfp-close:hover, .mfp-close:focus {
      opacity: 1; }
    .mfp-close:active {
      top: 1px; }
  
  .mfp-close-btn-in .mfp-close {
    color: #333333; }
  
  .mfp-image-holder .mfp-close,
  .mfp-iframe-holder .mfp-close {
    color: white;
    right: -6px;
    text-align: right;
    padding-right: 6px;
    width: 100%; }
  
  .mfp-counter {
    position: absolute;
    top: 0;
    right: 0;
    color: #cccccc;
    font-size: 12px;
    line-height: 18px; }
  
  .mfp-arrow {
    position: absolute;
    opacity: 0.65;
    margin: 0;
    top: 50%;
    margin-top: -55px;
    padding: 0;
    width: 90px;
    height: 110px;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); }
  
  .mfp-arrow:active {
    margin-top: -54px; }
  
  .mfp-arrow:hover,
  .mfp-arrow:focus {
    opacity: 1; }
  
  .mfp-arrow:before, .mfp-arrow:after,
  .mfp-arrow .mfp-b,
  .mfp-arrow .mfp-a {
    content: '';
    display: block;
    width: 0;
    height: 0;
    position: absolute;
    left: 0;
    top: 0;
    margin-top: 35px;
    margin-left: 35px;
    border: medium inset transparent; }
  .mfp-arrow:after,
  .mfp-arrow .mfp-a {
    border-top-width: 13px;
    border-bottom-width: 13px;
    top: 8px; }
  .mfp-arrow:before,
  .mfp-arrow .mfp-b {
    border-top-width: 21px;
    border-bottom-width: 21px; }
  
  .mfp-arrow-left {
    left: 0; }
    .mfp-arrow-left:after,
    .mfp-arrow-left .mfp-a {
      border-right: 17px solid white;
      margin-left: 31px; }
    .mfp-arrow-left:before,
    .mfp-arrow-left .mfp-b {
      margin-left: 25px;
      border-right: 27px solid #3f3f3f; }
  
  .mfp-arrow-right {
    right: 0; }
    .mfp-arrow-right:after,
    .mfp-arrow-right .mfp-a {
      border-left: 17px solid white;
      margin-left: 39px; }
    .mfp-arrow-right:before,
    .mfp-arrow-right .mfp-b {
      border-left: 27px solid #3f3f3f; }
  
  .mfp-iframe-holder {
    padding-top: 40px;
    padding-bottom: 40px; }
  
  .mfp-iframe-holder .mfp-content {
    line-height: 0;
    width: 100%;
    max-width: 900px; }
  
  .mfp-iframe-scaler {
    width: 100%;
    height: 0;
    overflow: hidden;
    padding-top: 56.25%; }
  
  .mfp-iframe-scaler iframe {
    position: absolute;
    display: block;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: black; }
  
  .mfp-iframe-holder .mfp-close {
    top: -40px; }
  
  /* Main image in popup */
  img.mfp-img {
    width: auto;
    max-width: 100%;
    height: auto;
    display: block;
    line-height: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    padding: 40px 0 40px;
    margin: 0 auto; }
  
  /* The shadow behind the image */
  .mfp-figure:after {
    content: '';
    position: absolute;
    left: 0;
    top: 40px;
    bottom: 40px;
    display: block;
    right: 0;
    width: auto;
    height: auto;
    z-index: -1;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: #444444; }
  
  .mfp-figure {
    line-height: 0; }
  
  .mfp-bottom-bar {
    margin-top: -36px;
    position: absolute;
    top: 100%;
    left: 0;
    width: 100%;
    cursor: auto; }
  
  .mfp-title {
    text-align: left;
    line-height: 18px;
    color: #f3f3f3;
    word-wrap: break-word;
    padding-right: 36px; }
  
  .mfp-figure small {
    color: #bdbdbd;
    display: block;
    font-size: 12px;
    line-height: 14px; }
  
  .mfp-image-holder .mfp-content {
    max-width: 100%; }
  
  .mfp-gallery .mfp-image-holder .mfp-figure {
    cursor: pointer; }
  
  @media screen and (max-width: 800px) and (orientation: landscape), screen and (max-height: 300px) {
    /**
     * Remove all paddings around the image on small screen
     */
    .mfp-img-mobile .mfp-image-holder {
      padding-left: 0;
      padding-right: 0; }
  
    .mfp-img-mobile img.mfp-img {
      padding: 0; }
  
    /* The shadow behind the image */
    .mfp-img-mobile .mfp-figure:after {
      top: 0;
      bottom: 0; }
  
    .mfp-img-mobile .mfp-bottom-bar {
      background: rgba(0, 0, 0, 0.6);
      bottom: 0;
      margin: 0;
      top: auto;
      padding: 3px 5px;
      position: fixed;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      box-sizing: border-box; }
  
    .mfp-img-mobile .mfp-bottom-bar:empty {
      padding: 0; }
  
    .mfp-img-mobile .mfp-counter {
      right: 5px;
      top: 3px; }
  
    .mfp-img-mobile .mfp-close {
      top: 0;
      right: 0;
      width: 35px;
      height: 35px;
      line-height: 35px;
      background: rgba(0, 0, 0, 0.6);
      position: fixed;
      text-align: center;
      padding: 0; }
  
    .mfp-img-mobile .mfp-figure small {
      display: inline;
      margin-left: 5px; } }
  @media all and (max-width: 900px) {
    .mfp-arrow {
      -webkit-transform: scale(0.75);
      transform: scale(0.75); }
  
    .mfp-arrow-left {
      -webkit-transform-origin: 0;
      transform-origin: 0; }
  
    .mfp-arrow-right {
      -webkit-transform-origin: 100%;
      transform-origin: 100%; }
  
    .mfp-container {
      padding-left: 6px;
      padding-right: 6px; } }
  .mfp-ie7 .mfp-img {
    padding: 0; }
  .mfp-ie7 .mfp-bottom-bar {
    width: 600px;
    left: 50%;
    margin-left: -300px;
    margin-top: 5px;
    padding-bottom: 5px; }
  .mfp-ie7 .mfp-container {
    padding: 0; }
  .mfp-ie7 .mfp-content {
    padding-top: 44px; }
  .mfp-ie7 .mfp-close {
    top: 0;
    right: 0;
    padding-top: 0; }
  
  //Magnific overrides
  .mfp-image img{
    background: white;
  }
  .mfp-figure:after {
    background: white;
  }
  
  /*
   * Off Canvas navbar toggle right
   * --------------------------------------------------
   */
  
  @media screen and (max-width: 768px) {
    .row-offcanvas .collapsing {
    -webkit-transition: none 0;
      -moz-transition: none 0;
      transition: none 0;
    }
   .row-offcanvas .navbar {
    position: absolute;
    z-index: 2;
      right:0;
      height:100%;
      width:55px;
      border:0;
      background-color:transparent;
    }
    .row-offcanvas .navbar-toggle {
      margin-right: 5px;
      margin-left: 5px;
    }
    .row-offcanvas {
      position: relative;
    }
    .row-offcanvas-right.active .navbar {
    position: absolute;
    z-index: 2;
      right: -28.4%;
      width:40%;
      background-color:#eee;
      border:0 solid #ddd;
      border-left-width:1px;
    }
    .row-offcanvas-right.active {
      left: -30%;
    }
    .row-offcanvas-right.active .navbar-collapse {
      position: relative;
      width: 100%;
    }
    .row-offcanvas .content {
    /*width:calc(100% - 60px);*/
    }
  }
  </style>
</head>
<body>
<div id="wrap">
<div class="container">
<div class="row row-offcanvas row-offcanvas-right">
<div class="contents col-xs-12 col-md-10">
<h1>Evaluate simulation</h1>
<p>This report evaluates the simulation results.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">library('GenomicRanges')</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: BiocGenerics
## Loading required package: parallel
## 
## Attaching package: 'BiocGenerics'
## 
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following object is masked from 'package:stats':
## 
##     xtabs
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, as.vector, cbind,
##     colnames, do.call, duplicated, eval, evalq, Filter, Find, get,
##     intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rep.int, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unlist, unsplit
## 
## Loading required package: S4Vectors
## Loading required package: stats4
## Loading required package: IRanges
## Loading required package: GenomeInfoDb
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">library('TxDb.Hsapiens.UCSC.hg19.knownGene')</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: GenomicFeatures
## Loading required package: AnnotationDbi
## Loading required package: Biobase
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
## 
## 
## Attaching package: 'AnnotationDbi'
## 
## The following object is masked from 'package:GenomeInfoDb':
## 
##     species
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">library('knitr')
library('devtools')
load('../simulation_info.Rdata')
load('../derAnalysis/run2-v1.0.10/fullRegions.Rdata')
names(fullRegions) <- seq_len(length(fullRegions))</code></pre>
</div>
<h1>Results</h1>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Find exons
txdb <- keepSeqlevels(TxDb.Hsapiens.UCSC.hg19.knownGene, 'chr22')
txinfo <- select(txdb, keys = chosen$ucsckg_id, columns = columns(txdb), keytype = 'TXNAME')

## Buiild GRangesList with exons grouped by transcript
tx <- split(GRanges(seqnames = txinfo$EXONCHROM, IRanges(start = txinfo$EXONSTART, end = txinfo$EXONEND), strand = txinfo$EXONSTRAND), txinfo$TXNAME)
tx <- tx[match(chosen$ucsckg_id, names(tx))]

## Find overlaps with DERs
ctov <- countOverlaps(tx, fullRegions)

## Check result with FWER sig 
fwer <-  fullRegions[fullRegions$significantFWER == 'TRUE']
min.ov <- min(min(width(fwer)), min(width(tx)))
ctov.fwer <- countOverlaps(tx, fwer, minoverlap =  min.ov)

## Use appropriate set
ctov.use <- if (identical(ctov > 0, ctov.fwer > 0)) ctov else ctov.fwer</code></pre>
</div>
<h2>Transcripts / genes vs DERs</h2>
<h3>Overview</h3>
<p>Table showing the results between whether the transcript was set to be differentially expressed (DE) and if it overlaps (minimum 1 bp) any candidate DER.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">addmargins(table('DE status' = chosen$DE, 'Overlaps DER' = ctov > 0))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##          Overlaps DER
## DE status FALSE TRUE Sum
##     FALSE    25   23  48
##     TRUE      5   43  48
##     Sum      30   66  96
</code></pre>
</div>
<p>The results are not the same using a minimum overlap of 13 bp between transcripts and candidate DERs with a FWER adjusted p-value &lt; 0.05. Thus, we will use only the DERs with FWER adjusted p-value &lt; 0.05.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">addmargins(table('DE status' = chosen$DE, 'Overlaps DER (sig FWER)' = ctov.fwer > 0))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##          Overlaps DER (sig FWER)
## DE status FALSE TRUE Sum
##     FALSE    37   11  48
##     TRUE      6   42  48
##     Sum      43   53  96
</code></pre>
</div>
<p>At a finer level, there is a difference in the number of exons per transcript overlapping all candidate DERs vs the DERs with FWER adjusted p-value &lt; 0.05.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Verify things are working properly
# table(countOverlaps(tx, fullRegions, minoverlap = min.ov) - ctov.fwer)

## Difference in overlaps found
table(ctov - ctov.fwer)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## 
##   0   1   2   3   4   6   7   8   9  11  12  15  16  17  53  64 121 125 
##  48  11   2   9   4   5   2   1   2   3   2   1   1   1   1   1   1   1
</code></pre>
</div>
<h3>By case</h3>
<p>We can separate the transcripts by their experiment setup case. That is, whether its from a gene with:</p>
<ul>
<li>a single transcript
<ul>
<li>set to be DE: <em>singleDE</em></li>
<li>set not to be DE: <em>singleNotDe</em></li>
</ul></li>
<li>two transcripts
<ul>
<li>with both set to be DE: <em>bothDE</em></li>
<li>with only one transcript set to be DE: <em>oneDE</em></li>
<li>with both set not to be DE: <em>noneDE</em></li>
</ul></li>
</ul>
<p>Then compare against the results where</p>
<ul>
<li><em>success.DE</em> means that the transcript was set to be DE and overlaps a DER (true positive)</li>
<li><em>failed.DE</em> means that the transcript was set to be DE and doesn't overlap a DER (false negative)</li>
<li><em>success.DER</em> means that the transcript was set not to de DE and doesn't overlap a DER (true negative)</li>
<li><em>failed.DER</em> means that the transcript was set not to be DE and does overlap a DER (false positive)</li>
</ul>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Indexes
idx <- list(success = list(de = chosen$DE & ctov.use > 0, der = !chosen$DE & !ctov.use > 0), failed = list(de = chosen$DE & !ctov.use > 0, der = !chosen$DE & ctov.use > 0))
idx <- lapply(idx, function(x) { lapply(x, which) })

## Classify results
chosen$result <- 'Success.DE'
chosen$result[ idx$success$der ] <- 'Success.DER'
chosen$result[ idx$failed$de ] <- 'Failed.DE'
chosen$result[ idx$failed$der ] <- 'Failed.DER'

## Overall summary
kable(table(chosen$case, chosen$result), format = 'html')</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Failed.DE
</th>
<th style="text-align:right;">
Failed.DER
</th>
<th style="text-align:right;">
Success.DE
</th>
<th style="text-align:right;">
Success.DER
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bothDE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
noneDE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
singleNotDE
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
12
</td>
</tr>
</tbody>
</table>
</div>
<h3>Failed.DE (false negative)</h3>
<p>The 6 <em>Failed.DE</em> cases (false negatives) are mostly short single transcript genes (one exon only) where 5 were set to have low expression on one group, normal on the other two.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Successful cases
success.de <- tx[ idx$success$de  ]
success.der <- tx[  idx$success$der ]

## What happened with the 2 txs set to be DE that were not picked up?
failed.de <- tx[ idx$failed$de  ]

## They are short transcripts
kable(chosen[idx$failed$de, ], format = 'html')</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
tx_idx
</th>
<th style="text-align:right;">
tx_n
</th>
<th style="text-align:right;">
tx_i
</th>
<th style="text-align:left;">
gene_id
</th>
<th style="text-align:left;">
ucsckg_id
</th>
<th style="text-align:right;">
fasta_i
</th>
<th style="text-align:left;">
DE
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
group3
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:right;">
readspertx
</th>
<th style="text-align:right;">
mean1
</th>
<th style="text-align:right;">
mean2
</th>
<th style="text-align:right;">
mean3
</th>
<th style="text-align:left;">
case
</th>
<th style="text-align:left;">
result
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100126318
</td>
<td style="text-align:left;">
uc021wmk.1
</td>
<td style="text-align:right;">
181
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
11
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100302149
</td>
<td style="text-align:left;">
uc021wrh.1
</td>
<td style="text-align:right;">
796
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:right;">
66
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
17
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100422998
</td>
<td style="text-align:left;">
uc021wnn.1
</td>
<td style="text-align:right;">
317
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
86
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
22
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100500833
</td>
<td style="text-align:left;">
uc010gvn.2
</td>
<td style="text-align:right;">
347
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
high
</td>
<td style="text-align:right;">
110
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
88
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
24
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100500901
</td>
<td style="text-align:left;">
uc021wny.1
</td>
<td style="text-align:right;">
423
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
126
</td>
<td style="text-align:right;">
272
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
3761
</td>
<td style="text-align:left;">
uc003avt.2
</td>
<td style="text-align:right;">
605
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:right;">
2075
</td>
<td style="text-align:right;">
830
</td>
<td style="text-align:right;">
830
</td>
<td style="text-align:right;">
830
</td>
<td style="text-align:right;">
415
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DE
</td>
</tr>
</tbody>
</table>
</div>
<p>However, 2 similar cases with short transcripts were successfully detected. So it's likely that a lower F-stat cutoff would have picked up these false negative cases.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## However there are other short transcripts that were picked up
kable(chosen[ idx$success$de[sum(width(success.de)) <= 110], ], format = 'html')</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
tx_idx
</th>
<th style="text-align:right;">
tx_n
</th>
<th style="text-align:right;">
tx_i
</th>
<th style="text-align:left;">
gene_id
</th>
<th style="text-align:left;">
ucsckg_id
</th>
<th style="text-align:right;">
fasta_i
</th>
<th style="text-align:left;">
DE
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
group3
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:right;">
readspertx
</th>
<th style="text-align:right;">
mean1
</th>
<th style="text-align:right;">
mean2
</th>
<th style="text-align:right;">
mean3
</th>
<th style="text-align:left;">
case
</th>
<th style="text-align:left;">
result
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100302118
</td>
<td style="text-align:left;">
uc021wls.1
</td>
<td style="text-align:right;">
127
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
high
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Success.DE
</td>
</tr>
<tr>
<td style="text-align:left;">
23
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
100500860
</td>
<td style="text-align:left;">
uc021wlo.1
</td>
<td style="text-align:right;">
115
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
low
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
88
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:left;">
singleDE
</td>
<td style="text-align:left;">
Success.DE
</td>
</tr>
</tbody>
</table>
</div>
<p>More info:</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">width(failed.de)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## IntegerList of length 6
## [["uc021wmk.1"]] 78
## [["uc021wrh.1"]] 66
## [["uc021wnn.1"]] 86
## [["uc010gvn.2"]] 110
## [["uc021wny.1"]] 58
## [["uc003avt.2"]] 220 1844
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">width(tx[ idx$success$de[sum(width(success.de)) <= 110] ])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## IntegerList of length 2
## [["uc021wls.1"]] 78
## [["uc021wlo.1"]] 88
</code></pre>
</div>
<h3>Failed.DER (false positive)</h3>
<p>Out of the 11 <em>Failed.DER</em> transcripts (false positives), 11 of them are from the <em>oneDE</em> case. You could then argue that they are really not false positives. However, 0 and 0 transcripts are from the <em>noneDE</em> and <em>singleNotDE</em> cases respectively which would be the truly false positives.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## What happened with those set to be not DE but overlap DERs?
failed.der <- tx[ idx$failed$der ]

kable(chosen[idx$failed$der, ], format = 'html')</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
tx_idx
</th>
<th style="text-align:right;">
tx_n
</th>
<th style="text-align:right;">
tx_i
</th>
<th style="text-align:left;">
gene_id
</th>
<th style="text-align:left;">
ucsckg_id
</th>
<th style="text-align:right;">
fasta_i
</th>
<th style="text-align:left;">
DE
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
group3
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:right;">
readspertx
</th>
<th style="text-align:right;">
mean1
</th>
<th style="text-align:right;">
mean2
</th>
<th style="text-align:right;">
mean3
</th>
<th style="text-align:left;">
case
</th>
<th style="text-align:left;">
result
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
26
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
100130717
</td>
<td style="text-align:left;">
uc011agh.3
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
650
</td>
<td style="text-align:right;">
260
</td>
<td style="text-align:right;">
260
</td>
<td style="text-align:right;">
260
</td>
<td style="text-align:right;">
260
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
37
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
10738
</td>
<td style="text-align:left;">
uc010gwn.3
</td>
<td style="text-align:right;">
467
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
1488
</td>
<td style="text-align:right;">
595
</td>
<td style="text-align:right;">
595
</td>
<td style="text-align:right;">
595
</td>
<td style="text-align:right;">
595
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
47
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
128977
</td>
<td style="text-align:left;">
uc002zpi.3
</td>
<td style="text-align:right;">
84
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
1558
</td>
<td style="text-align:right;">
623
</td>
<td style="text-align:right;">
623
</td>
<td style="text-align:right;">
623
</td>
<td style="text-align:right;">
623
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
49
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
129138
</td>
<td style="text-align:left;">
uc003auc.3
</td>
<td style="text-align:right;">
576
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
2149
</td>
<td style="text-align:right;">
860
</td>
<td style="text-align:right;">
860
</td>
<td style="text-align:right;">
860
</td>
<td style="text-align:right;">
860
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
98
</td>
<td style="text-align:right;">
206
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
25817
</td>
<td style="text-align:left;">
uc003bio.4
</td>
<td style="text-align:right;">
836
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
2652
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
121
</td>
<td style="text-align:right;">
266
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
339669
</td>
<td style="text-align:left;">
uc003aqe.3
</td>
<td style="text-align:right;">
538
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
1049
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
132
</td>
<td style="text-align:right;">
283
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
3976
</td>
<td style="text-align:left;">
uc011aks.2
</td>
<td style="text-align:right;">
379
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
3987
</td>
<td style="text-align:right;">
1595
</td>
<td style="text-align:right;">
1595
</td>
<td style="text-align:right;">
1595
</td>
<td style="text-align:right;">
1595
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
150
</td>
<td style="text-align:right;">
321
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
4689
</td>
<td style="text-align:left;">
uc003apz.4
</td>
<td style="text-align:right;">
533
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
1646
</td>
<td style="text-align:right;">
658
</td>
<td style="text-align:right;">
658
</td>
<td style="text-align:right;">
658
</td>
<td style="text-align:right;">
658
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
175
</td>
<td style="text-align:right;">
400
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
6523
</td>
<td style="text-align:left;">
uc003amc.3
</td>
<td style="text-align:right;">
457
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
5061
</td>
<td style="text-align:right;">
2024
</td>
<td style="text-align:right;">
2024
</td>
<td style="text-align:right;">
2024
</td>
<td style="text-align:right;">
2024
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
190
</td>
<td style="text-align:right;">
417
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
6948
</td>
<td style="text-align:left;">
uc003air.2
</td>
<td style="text-align:right;">
407
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
2006
</td>
<td style="text-align:right;">
802
</td>
<td style="text-align:right;">
802
</td>
<td style="text-align:right;">
802
</td>
<td style="text-align:right;">
802
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
<tr>
<td style="text-align:left;">
192
</td>
<td style="text-align:right;">
421
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
7122
</td>
<td style="text-align:left;">
uc010grr.2
</td>
<td style="text-align:right;">
91
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:left;">
normal
</td>
<td style="text-align:right;">
1720
</td>
<td style="text-align:right;">
688
</td>
<td style="text-align:right;">
688
</td>
<td style="text-align:right;">
688
</td>
<td style="text-align:right;">
688
</td>
<td style="text-align:left;">
oneDE
</td>
<td style="text-align:left;">
Failed.DER
</td>
</tr>
</tbody>
</table>
</div>
<h4>Other strand</h4>
<p>In some simulations, we found what seemed to be false positive transcripts but turned out to overlap DERs in regions where there are exons on both the positive and negative strands and at least one of the exons was set to be DE.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Most of the truly false positive transcripts don't overlap other transcripts
i <- intersect(idx$failed$der, which(chosen$case == 'noneDE'))
table(countOverlaps(tx[i], tx[-i]))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## < table of extent 0 >
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## They are not short
width(tx[i])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## IntegerList of length 0
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## To explore regions with derfinderReport
sort(subjectHits(findOverlaps(tx[i], fullRegions)))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## integer(0)
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## DERs
# fullRegions[subjectHits(findOverlaps(tx[i], fullRegions))]</code></pre>
</div>
<p><em>Skipped the following section due to the absence of such cases in this particular simulation</em></p>
<p>As it can be seen below, 0 apparent false positive transcripts from the <em>noneDE</em> case overlap (when strand is not taken into account) genes where at least one of two transcripts was set to be DE.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">kable(chosen[ subjectHits(findOverlaps(tx[i], tx[-i], ignore.strand = TRUE)), ], format = 'html')</code></pre>
<table>
<thead>
<tr>
<th style="text-align:right;">
tx_idx
</th>
<th style="text-align:right;">
tx_n
</th>
<th style="text-align:right;">
tx_i
</th>
<th style="text-align:left;">
gene_id
</th>
<th style="text-align:left;">
ucsckg_id
</th>
<th style="text-align:right;">
fasta_i
</th>
<th style="text-align:left;">
DE
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
group3
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:right;">
readspertx
</th>
<th style="text-align:right;">
mean1
</th>
<th style="text-align:right;">
mean2
</th>
<th style="text-align:right;">
mean3
</th>
<th style="text-align:left;">
case
</th>
<th style="text-align:left;">
result
</th>
</tr>
</thead>
<tbody>
<tr>
</tr>
</tbody>
</table>
</div>
<h3>Gene level</h3>
<p>For each gene, if at least one transcript is set to be DE then we consider the gene to be DE. Then, we check if the gene overlaps at least one DER.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">gene <- data.frame(gene_id = unique(chosen$gene_id))
gene$DE <- sapply(gene$gene_id, function(x) { any(chosen$DE[chosen$gene_id == x])  })
gene$case <- sapply(gene$gene_id, function(x) { unique(chosen$case[chosen$gene_id == x])  })
gene$overlaps <- sapply(gene$gene_id, function(x) { sum(ctov.use[ chosen$ucsckg_id[chosen$gene_id == x] ]) })
gene$overlap <- gene$overlaps > 0

## Results between DE status and overlapping at least 1 DER at the gene level
addmargins(table('DE status' = gene$DE, 'Overlaps DER' = gene$overlap))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##          Overlaps DER
## DE status FALSE TRUE Sum
##     FALSE    24    0  24
##     TRUE      6   30  36
##     Sum      30   30  60
</code></pre>
</div>
<h3>Conclusions</h3>
<p>The results from the simulation are promising as most transcripts were correctly classified as differentially expressed or not by <code>derfinder</code>.</p>
<p>The majority of the false negative cases involved short single transcript genes with one group having low expression relative to the other two. These cases could potentially be eliminated by lowering the F-statistic threshold used in the <code>derfinder</code> analysis.</p>
<p>In some simulations there are some apparent false positives which are due to transcripts on one strand set not to be DE overlapping transcripts from the other strand set to be DE. This situation could be solved with strand-specific RNA-seq data and running <code>derfinder</code> for each strand separately.</p>
<h3>Extra</h3>
<p>Minimum number of reads per transcript as well as per sample.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Distribution of the minimum number of reads per transcript
summary(apply(readmat, 1, min))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     1.0   213.2   453.5   625.5   783.0  2747.0
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Distribution of the minimum number of reads per sample
summary(apply(readmat, 2, min))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   1.000   7.000   9.000   9.067  11.000  19.000
</code></pre>
</div>
<p>The minimum number of reads per transcript for a given sample is 1.</p>
<h2>DERs vs segments</h2>
<p>Next we can evaluate the simulation by classifying the exonic segments as whether they should be DE or not. Then, we can find if the DERs overlap such segments. We would expect that the DERs with a FWER adjusted p-value &lt; 0.05 would only overlap segments that were set to be DE.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">segments <- GRangesList(lapply(gene$gene_id, function(x) {
    i <- chosen$ucsckg_id[ chosen$gene_id == x]
    
    ## Find segments
    segs <- disjoin(unlist(tx[i]))
    ov <- findOverlaps(segs, tx[i])
    
    ## Find DE status per segment
    segs$DE <- as.vector(tapply(subjectHits(ov), queryHits(ov), function(y) {
        any(chosen$DE[ chosen$gene_id == x])
    }))
    
    ## Finish
    return(segs)
}))
names(segments) <- gene$gene_id
segs <- unlist(segments)


## Do DERs overlap segments that are set to be DE?
der.ov <- findOverlaps(fullRegions, segs)
fullRegions$overlap <- sapply(seq_len(length(fullRegions)), function(x) {
    y <- which(queryHits(der.ov) == x)
    if(length(y) == 0) return(NA)
    any(segs$DE[ subjectHits(der.ov)[y] ])
})</code></pre>
</div>
<h3>Segments per DER</h3>
<p>We can check how many segments each DER overlaps. Ideally they should all overlap at least one segment, but there are some cases where this could not happen (6 in this case). Mostly because of small mismatches between the transcripts and the actual mRNA used in the simulation.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Do DERs overlap at least one segment?
table(countOverlaps(fullRegions, segs))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## 
##   0   1   2   3   4 
##   6 451   9   1   2
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Widths of DERs not overlapping any segment
width(fullRegions[is.na(fullRegions$overlap)])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] 39  4  3  1  1  1
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Do these DERs have a significant FWER adjusted p-value?
table(fullRegions$significantFWER[is.na(fullRegions$overlap)])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## 
##  TRUE FALSE 
##     0     6
</code></pre>
</div>
<h3>DER correctness</h3>
<p>However, the main result is whether the DERs overlap segments expected to be DE. Note that for this comparison, DERs are unstranded and could potentially overlap two segments from different strands where only one of them was set to be DE.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Check by whether the DER has a FWER adjusted p-value < 0.05
addmargins(table('Overlaps a DE segment' = fullRegions$overlap, 'FWER p-value < 0.05' = fullRegions$significantFWER))</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##                      FWER p-value < 0.05
## Overlaps a DE segment TRUE FALSE Sum
##                 FALSE    0    30  30
##                 TRUE   122   311 433
##                 Sum    122   341 463
</code></pre>
</div>
<p>Regardless of whether the DER p-value, we see that 6.48 percent of the DERs overlapping at least one segment, incorrectly overlap a segment set not to be DE.</p>
<h3>Conclusions</h3>
<p>None of the DERs with a FWER adjusted p-value &lt; 0.05 overlaps a segment set not to be DE, which is what we would expect.</p>
<h1>Reproducibility</h1>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">## Reproducibility info
Sys.time()</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] "2014-12-04 17:39:35 EST"
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">proc.time()</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##    user  system elapsed 
##  18.075   0.235  18.318
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-up"></span> R source
</button>
<pre style="display:none"><code class="source r">options(width = 120)
session_info()</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Session info-----------------------------------------------------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  setting  value                                             
##  version  R Under development (unstable) (2014-11-01 r66923)
##  system   x86_64, darwin10.8.0                              
##  ui       X11                                               
##  language (EN)                                              
##  collate  en_US.UTF-8                                       
##  tz       America/New_York
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Packages---------------------------------------------------------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  package                           * version  date       source                                   
##  AnnotationDbi                     * 1.29.10  2014-11-22 Bioconductor                             
##  base64enc                           0.1.2    2014-06-26 CRAN (R 3.2.0)                           
##  BatchJobs                           1.4      2014-09-24 CRAN (R 3.2.0)                           
##  BBmisc                              1.7      2014-06-21 CRAN (R 3.2.0)                           
##  Biobase                           * 2.27.0   2014-10-14 Bioconductor                             
##  BiocGenerics                      * 0.13.2   2014-11-18 Bioconductor                             
##  BiocParallel                        1.1.9    2014-11-24 Bioconductor                             
##  biomaRt                             2.23.5   2014-11-22 Bioconductor                             
##  Biostrings                          2.35.5   2014-11-24 Bioconductor                             
##  bitops                              1.0.6    2013-08-17 CRAN (R 3.2.0)                           
##  brew                                1.0.6    2011-04-13 CRAN (R 3.2.0)                           
##  checkmate                           1.5.0    2014-10-19 CRAN (R 3.2.0)                           
##  codetools                           0.2.9    2014-08-21 CRAN (R 3.2.0)                           
##  DBI                                 0.3.1    2014-09-24 CRAN (R 3.2.0)                           
##  devtools                          * 1.6.1    2014-10-07 CRAN (R 3.2.0)                           
##  digest                              0.6.4    2013-12-03 CRAN (R 3.2.0)                           
##  evaluate                            0.5.5    2014-04-29 CRAN (R 3.2.0)                           
##  fail                                1.2      2013-09-19 CRAN (R 3.2.0)                           
##  foreach                             1.4.2    2014-04-11 CRAN (R 3.2.0)                           
##  formatR                             1.0      2014-08-25 CRAN (R 3.2.0)                           
##  GenomeInfoDb                      * 1.3.7    2014-11-15 Bioconductor                             
##  GenomicAlignments                   1.3.12   2014-11-27 Bioconductor                             
##  GenomicFeatures                   * 1.19.6   2014-11-04 Bioconductor                             
##  GenomicRanges                     * 1.19.18  2014-12-01 Bioconductor                             
##  htmltools                           0.2.6    2014-09-08 CRAN (R 3.2.0)                           
##  IRanges                           * 2.1.24   2014-12-01 Bioconductor                             
##  iterators                           1.0.7    2014-04-11 CRAN (R 3.2.0)                           
##  knitr                             * 1.7      2014-10-13 CRAN (R 3.2.0)                           
##  knitrBootstrap                      1.0.0    2014-11-03 Github (jimhester/knitrBootstrap@76c41f0)
##  markdown                            0.7.4    2014-08-24 CRAN (R 3.2.0)                           
##  RCurl                               1.95.4.3 2014-07-29 CRAN (R 3.2.0)                           
##  rmarkdown                         * 0.3.3    2014-09-17 CRAN (R 3.2.0)                           
##  Rsamtools                           1.19.11  2014-11-26 Bioconductor                             
##  RSQLite                             1.0.0    2014-10-25 CRAN (R 3.2.0)                           
##  rstudioapi                          0.1      2014-03-27 CRAN (R 3.2.0)                           
##  rtracklayer                         1.27.6   2014-11-26 Bioconductor                             
##  S4Vectors                         * 0.5.11   2014-11-24 Bioconductor                             
##  sendmailR                           1.2.1    2014-09-21 CRAN (R 3.2.0)                           
##  stringr                             0.6.2    2012-12-06 CRAN (R 3.2.0)                           
##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.0.0    2014-09-26 Bioconductor                             
##  XML                                 3.98.1.1 2013-06-20 CRAN (R 3.2.0)                           
##  XVector                           * 0.7.3    2014-11-24 Bioconductor                             
##  yaml                                2.1.13   2014-06-12 CRAN (R 3.2.0)                           
##  zlibbioc                            1.13.0   2014-10-14 Bioconductor
</code></pre>
</div>
</div>
</div>
<div class="navbar navbar-fixed-bottom navbar-inverse">
<div class="container">
<div class="navbar-header">
<button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-responsive-collapse">
<span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span>
</button>
</div>
<div id="bottom-navbar" class="navbar-collapse collapse navbar-responsive-collapse">
<ul class="nav navbar-nav navbar-right">
<li class="nav">
<p class="navbar-text">
Toggle
</p>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Languages
</li>
<li class="active">
<a href="#" class="toggle-global source R" type="source.R">R</a>
</li>
<li>
<a href="#" type="all-source" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Output <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Type
</li>
<li class="active">
<a href="#" class="toggle-global message" type="message">message</a>
</li>
<li class="active">
<a href="#" class="toggle-global output" type="output">output</a>
</li>
<li>
<a href="#" type="all-output" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="active">
<a href="#" type="figure" class="toggle-global">Figures</a>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Bootstrap theme <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="theme-switch" title="Default" href="#">Default</a>
</li>
<li>
<a class="theme-switch" title="Amelia" href="#">Amelia</a>
</li>
<li>
<a class="theme-switch" title="Cerulean" href="#">Cerulean</a>
</li>
<li>
<a class="theme-switch" title="Cosmo" href="#">Cosmo</a>
</li>
<li>
<a class="theme-switch" title="Cyborg" href="#">Cyborg</a>
</li>
<li>
<a class="theme-switch" title="Journal" href="#">Journal</a>
</li>
<li>
<a class="theme-switch" title="Flatly" href="#">Flatly</a>
</li>
<li>
<a class="theme-switch" title="Readable" href="#">Readable</a>
</li>
<li>
<a class="theme-switch" title="Simplex" href="#">Simplex</a>
</li>
<li>
<a class="theme-switch" title="Slate" href="#">Slate</a>
</li>
<li>
<a class="theme-switch" title="Spacelab" href="#">Spacelab</a>
</li>
<li>
<a class="theme-switch" title="United" href="#">United</a>
</li>
</ul>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code highlighting <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="highlight-switch" title="HighlightJs" href="#">HighlightJs</a>
</li>
<li>
<a class="highlight-switch" title="Dark" href="#">Dark</a>
</li>
<li>
<a class="highlight-switch" title="Far" href="#">Far</a>
</li>
<li>
<a class="highlight-switch" title="Idea" href="#">Idea</a>
</li>
<li>
<a class="highlight-switch" title="Sunburst" href="#">Sunburst</a>
</li>
<li>
<a class="highlight-switch" title="Zenburn" href="#">Zenburn</a>
</li>
<li>
<a class="highlight-switch" title="Visual Studio" href="#">Visual Studio</a>
</li>
<li>
<a class="highlight-switch" title="Ascetic" href="#">Ascetic</a>
</li>
<li>
<a class="highlight-switch" title="Magula" href="#">Magula</a>
</li>
<li>
<a class="highlight-switch" title="Github" href="#">Github</a>
</li>
<li>
<a class="highlight-switch" title="Google Code" href="#">Google Code</a>
</li>
<li>
<a class="highlight-switch" title="Brown Paper" href="#">Brown Paper</a>
</li>
<li>
<a class="highlight-switch" title="School Book" href="#">School Book</a>
</li>
<li>
<a class="highlight-switch" title="IR Black" href="#">IR Black</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Dark" href="#">Solarized - Dark</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Light" href="#">Solarized - Light</a>
</li>
<li>
<a class="highlight-switch" title="Arta" href="#">Arta</a>
</li>
<li>
<a class="highlight-switch" title="Monokai" href="#">Monokai</a>
</li>
<li>
<a class="highlight-switch" title="Xcode" href="#">Xcode</a>
</li>
<li>
<a class="highlight-switch" title="Pojoaque" href="#">Pojoaque</a>
</li>
<li>
<a class="highlight-switch" title="Rainbow" href="#">Rainbow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow" href="#">Tomorrow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night" href="#">Tomorrow Night</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Bright" href="#">Tomorrow Night Bright</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Blue" href="#">Tomorrow Night Blue</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Eighties" href="#">Tomorrow Night Eighties</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" title="Default" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/amelia/bootstrap.min.css" title="Amelia" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cerulean/bootstrap.min.css" title="Cerulean" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cosmo/bootstrap.min.css" title="Cosmo" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cyborg/bootstrap.min.css" title="Cyborg" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/journal/bootstrap.min.css" title="Journal" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/flatly/bootstrap.min.css" title="Flatly" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/readable/bootstrap.min.css" title="Readable" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/simplex/bootstrap.min.css" title="Simplex" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/slate/bootstrap.min.css" title="Slate" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/spacelab/bootstrap.min.css" title="Spacelab" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/united/bootstrap.min.css" title="United" media="screen"></link><link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" title="HighlightJs" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/dark.min.css" title="Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/far.min.css" title="Far" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/idea.min.css" title="Idea" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/sunburst.min.css" title="Sunburst" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/zenburn.min.css" title="Zenburn" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/vs.min.css" title="Visual Studio" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ascetic.min.css" title="Ascetic" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/magula.min.css" title="Magula" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/github.min.css" title="Github" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/googlecode.min.css" title="Google Code" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/brown_paper.min.css" title="Brown Paper" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/school_book.min.css" title="School Book" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ir_black.min.css" title="IR Black" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_dark.min.css" title="Solarized - Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_light.min.css" title="Solarized - Light" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/arta.min.css" title="Arta" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/monokai.min.css" title="Monokai" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/xcode.min.css" title="Xcode" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/pojoaque.min.css" title="Pojoaque" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/rainbow.min.css" title="Rainbow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow.min.css" title="Tomorrow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night.min.css" title="Tomorrow Night" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-bright.min.css" title="Tomorrow Night Bright" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-blue.min.css" title="Tomorrow Night Blue" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-eighties.min.css" title="Tomorrow Night Eighties" media="screen"></link>
</div>
<div id="push">

</div>
<div id="footer">
<div class="container">
<p class="text-muted" id="credit">
Styled with <a href="https://github.com/jimhester/knitrBootstrap">knitrBootstrap</a>
</p>
</div>
</div>
<link rel="stylesheet" id="theme" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" media="screen"></link><link rel="stylesheet" id="highlight" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" media="screen"></link>
</div>
</body>
</html>
